### ISSUE TITLE
txt2img training error

### ISSUE THREAD

**Issue by mwmw7** [CreamyLong/stable-diffusion#10]
return func(*args, **kwargs)
  File "/home/toonsquare/PycharmProjects/toonsquare-creamylong-stable-diffusion/ldm/models/diffusion/ddpm.py", line 735, in get_input
    xc = batch[cond_key]
KeyError: 'caption'
`  def _sample_generator(self, start, end):
        for idx in range(start, end):
            file_name = self.file_list[idx]
            txt_name = self.txt_list[idx]
            f_ = open(txt_name, 'r')
            txt_ = f_.read()
            f_.close()
            image = cv2.imdecode(np.fromfile(file_name, dtype=np.uint8), 1)
            image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
            image = torch.from_numpy(image) / 255
            yield {"txt": txt_, "image": image}
`

**Comment by CreamyLong** [comment]
Sorry, I forget to upload the file for txt2img dataset.
ldm/data/custom_txt.py has been updated.

**Comment by mwmw7** [comment]
> Sorry, I forget to upload the file for txt2img dataset. ldm/data/custom_txt.py has been updated.

Thank you ^^

**Comment by mwmw7** [comment]
> > Sorry, I forget to upload the file for txt2img dataset. ldm/data/custom_txt.py has been updated.
> 
> Thank you! I'll give it a try right away.

Still got a some problem

error message
![image](https://github.com/CreamyLong/stable-diffusion/assets/1140031/af942198-047e-49e4-abc0-622ef6f88d22)

CustomBase
![image](https://github.com/CreamyLong/stable-diffusion/assets/1140031/f9d12753-7df8-4c3a-9f52-f5bbb7250eb6)


my test data set
![image](https://github.com/CreamyLong/stable-diffusion/assets/1140031/b1c04c11-f976-40e4-9221-0df1b636301c)


I thing this CustomBase needs img path and it's description text path, 
How should I organize my data?

**Comment by CreamyLong** [comment]
Yeah,  I find there are some minor mistakes in the function. I recommend that you fix it and then pull the request. 
The img_paths are the list of paths of images. The txt_paths are the txt file which includes the description for the corresponding image.  For example, the 1st of description in txt_paths  is for the 1st image path in img_paths

**Comment by gg22mm** [comment]
python main.py --base configs/latent-diffusion/txt2img/txt2img-sdv1.yaml 我的这个错就有点奇怪：
好象有错，但又没报错就结束了~~

ion_modeloutput_blocks111transformer_blocks0attn2to_out0bias', 'model_ema.diffusion_modeloutput_blocks111transformer_blocks0norm1weight', 'model_ema.diffusion_modeloutput_blocks111transformer_blocks0norm1bias', 'model_ema.diffusion_modeloutput_blocks111transformer_blocks0norm2weight', 'model_ema.diffusion_modeloutput_blocks111transformer_blocks0norm2bias', 'model_ema.diffusion_modeloutput_blocks111transformer_blocks0norm3weight', 'model_ema.diffusion_modeloutput_blocks111transformer_blocks0norm3bias', 'model_ema.diffusion_modeloutput_blocks111proj_outweight', 'model_ema.diffusion_modeloutput_blocks111proj_outbias', 'model_ema.diffusion_modelout0weight', 'model_ema.diffusion_modelout0bias', 'model_ema.diffusion_modelout2weight', 'model_ema.diffusion_modelout2bias', 'cond_stage_model.transformer.text_model.embeddings.position_ids']
----------end222222222---------
Monitoring val/loss_simple_ema as checkpoint metric.
Merged modelckpt-cfg:
{'target': 'pytorch_lightning.callbacks.ModelCheckpoint', 'params': {'dirpath': 'logs/2024-07-06T19-01-35_txt2img-sdv1/checkpoints', 'filename': '{epoch:06}', 'verbose': True, 'save_last': True, 'monitor': 'val/loss_simple_ema', 'save_top_k': 3}}
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
#### Data #####
train, CustomTrain, 35
validation, CustomTest, 35
accumulate_grad_batches = 1
Setting learning rate to 2.00e-06 = 1 (accumulate_grad_batches) * 1 (num_gpus) * 1 (batchsize) * 2.00e-06 (base_lr)
/home/data/miniconda3/envs/mygpt39/lib/python3.9/site-packages/pytorch_lightning/core/datamodule.py:423: LightningDeprecationWarning: DataModule.setup has already been called, so it will not be called again. In v1.6 this behavior will change to always call DataModule.setup.
  rank_zero_deprecation(

(mygpt39) [root@14c0e63c0174 stable-diffusion-master]#

**Comment by gg22mm** [comment]
> Yeah, I find there are some minor mistakes in the function. I recommend that you fix it and then pull the request. The img_paths are the list of paths of images. The txt_paths are the txt file which includes the description for the corresponding image. For example, the 1st of description in txt_paths is for the 1st image path in img_paths

我改了，不知道怎提交： https://github.com/gg22mm/stable-diffusion

运行： python main.py --base configs/latent-diffusion/txt2img/txt2img-sdv1_me.yaml

**Comment by CreamyLong** [comment]
> > Yeah, I find there are some minor mistakes in the function. I recommend that you fix it and then pull the request. The img_paths are the list of paths of images. The txt_paths are the txt file which includes the description for the corresponding image. For example, the 1st of description in txt_paths is for the 1st image path in img_paths
> 
> 我改了，不知道怎提交：[master...gg22mm:stable-diffusion:master](https://github.com/CreamyLong/stable-diffusion/compare/master...gg22mm:stable-diffusion:master)
> 
> 运行： python main.py --base configs/latent-diffusion/txt2img/txt2img-sdv1_me.yaml --train True

You could use the function of pull request. Anyway, I have updated the code as you improved. Thank you.
